sound/soc/sof/loader.c

```diff
@@ -142,6 +142,18 @@ static int ext_man_get_fw_version(struct snd_sof_dev *sdev,
 	return snd_sof_ipc_valid(sdev);
 }
 
+static int ext_man_get_windows(struct snd_sof_dev *sdev,
+			       const struct ext_man_elem_header *hdr)
+{
+	const struct sof_ipc_ext_data_hdr *w_ipc;
+	const struct ext_man_window *w;
+
+	w = container_of(hdr, struct ext_man_window, hdr);
+	w_ipc = (const struct sof_ipc_ext_data_hdr *)&w->ipc_window;
+
+	return get_ext_windows(sdev, w_ipc);
```

Comment:
`get_ext_windows()` allocates memory. Can it happen, that firmware images will contain an extended manifest but the firmware itself will still be "old" and it will send its window information with an IPC? If yes, then we have to free memory in `get_ext_windows()` before re-allocating it.

Reply:
Ok, but I won't use `free()`, rather `memcmp()`